#! /bin/sh /usr/share/dpatch/dpatch-run
## 40_nonsusescript.dpatch by Petter Reinholdtsen <pere@hungry.com>

Make sure the 'single' script is not inserted into runlevel S, and
avoid fixing the sequence number for a lot of scripts used in SuSe.

@DPATCH@
--- insserv-1.08.0.orig/insserv.c	2007-11-21 17:30:08.000000000 +0100
+++ insserv-1.08.0/insserv.c	2007-11-21 17:30:33.000000000 +0100
@@ -438,12 +438,14 @@
     return serv;
 }
 
+#ifdef SUSE
 static void setlsb(const char* const name)
 {
     serv_t * serv = findserv(name);
     if (serv)
 	serv->opts &= ~SERV_NOTLSB;
 }
+#endif /* SUSE */
 
 /*
  * This helps us to set none LSB conform scripts to required
@@ -1824,6 +1824,7 @@
 	/* main scanner for LSB comment in current script */
 	lsb = scan_script_defaults(d->d_name);
 
+#ifdef SUSE
 	/* Common script ... */
 	if (!strcmp(d->d_name, "halt")) {
 	    makeprov("halt",   d->d_name);
@@ -1920,6 +1921,7 @@
 	    rememberreq(serv, REQ_SHLD, "kbd");
 	    continue;
 	}
+#endif /* SUSE */
 
 	/*
 	 * Oops, no comment found, guess one
@@ -2206,6 +2208,7 @@
 
     expand_conf();
 
+#ifdef SUSE
     /*
      *  Set initial order of some services
      */
@@ -2207,6 +2210,7 @@
     setorder("serial",	10, false); setlsb("serial");
     setorder("gpm",	20, false); setlsb("gpm");
     setorder("boot.setup", 20, false);
+#endif /* SUSE */
 
     /*
      * Set virtual dependencies for already enabled none LSB scripts.
